src/Constraint/MultiConstraint.php

```diff
@@ -75,6 +77,19 @@ public function isDisjunctive()
         return !$this->conjunctive;
     }
 
+    public function compile($otherOperator)
+    {
+        if (!$this->conjunctive) {
+            return '('.implode(') || (',\array_map(function(ConstraintInterface $constraint) use ($otherOperator) {
+                return $constraint->compile($otherOperator);
```

Comment:
What if the inner constraints are not compilable ones ?

Reply:
People should not call `compile` on a custom constraint that does not implement the new interface.
I thought about letting PHP throw an exception.

Do you think we should handle such case (ie. Plugin inject Custom Constraint in Composer)?

Reply:
Well, that's exactly what you do here, as you don't check at all whether the inner constraints of the MultiConstraint are custom or no.

And people won't call it. Composer will. If it is possible to have a non-compilable MultiConstraint, we need to have a working way to bailout (not a PHP Fatal Error), so that the matcher can fallback to non-optimized mode. The compilation is just an optimization, so we should not fail hard when we cannot do it.

Reply:
I am not aware of any custom constraint implementations out in the wild.. Strictly speaking it would be good to check for the interface and fail if it is not compilable, but if this has an adverse effect performance wise we maybe rather just leave it as is and let people doing custom stuff figure out why they get a fatal..

Reply:
Fixed (catch the exception and fallback to class constraint match when a constraint is not compilable)